### ISSUE TITLE
No INIFile, AppImage fails to execute.

### ISSUE THREAD

**Issue by LordSicarious** [AM2R-Community-Developers/AM2RLauncher#19]
Tried installing on Linux Mint via the new 2.1 launcher. The AppImage was successfully created, however it cannot be executed. I suspect the output is related to some issue with my OpenGL install (version 1.4), but I haven't been able to identify any issues myself. Attempting to execute the file via terminal yields the following result:

```
***************************************
*     YoYo Games Linux Runner V1.3    *
***************************************	 
CommandLine: -game game.unx
ExeName= /tmp/.mount_AM2R.A9SieAp/usr/bin/AM2R
MemoryManager allocated: 17920

***************************************
*     YoYo Games Runner v1.0(999)[r32908]    *
***************************************	 
RunnerLoadGame: game.unx
RunnerLoadGame() - assets/game.unx
Checking if INIFile assets/options.ini Exists
!!!!No INI File assets/options.ini
Reading File assets/game.unx
Loaded File assets/game.unx(68872432)
IFF wad found
Get Resolution
Get Header Information
InitGMLFunctions
HighScore..SavePrePend /home/sean/.config/AM2R/ 
Game..Math..Graphic..Action..File..Resource..Interaction..3D..Particle..Misc..DS..Sound..Physics..Gamepad..GAMEPAD: Initialising Ubuntu support
GAMEPAD: 4 axis values (last) at 0xf682d660
GAMEPAD: 4 axis values (last) at 0xf682d640
GAMEPAD: 4 axis values (last) at 0xf682d620
GAMEPAD: 4 axis values (last) at 0xf682d600
Buffers..Networking..Shaders..InitPushFunctions...YoYo..Fini
ID_STRG
Create Window
Display Size(Pixels): 1920,1200
GameDisplayName=AM2R
Win #1
XF86VidModeExtension-Version 2.2
libGL error: No matching fbConfigs or visuals found
libGL error: failed to load driver: swrast
Only Singlebuffered Visual!
glX-Version 1.4
Segmentation fault (core dumped)

```

**Comment by Lojemiru** [comment]
What GPU are you using? Kind of wrong repository but given that Linux crashes for the game itself are pretty typical I'll just troubleshoot it here.

**Comment by LordSicarious** [comment]
EVGA NVidia Geforce GTX 970 FTW, I haven't had these issues elsewhere and have been running this GPU for ~5 years, so not sure why this in particular is causing issues. I can run 1.1 in Wine without issue, but I just found out about these cool community patches with a native linux launcher and wanted to give them a try.

**Comment by Lojemiru** [comment]
Alright. Open the `Options` tab in the AM2RLauncher and input this preload argument in the launch arguments field: `LD_PRELOAD=/usr/lib/i386-linux-gnu/libGL.so.1`.

**Comment by LordSicarious** [comment]
Presuming you mean settings/custom game environment variables, that's just causing different issues for me now. Now every time I try to run the game from the launcher, I'm getting errors that the launcher can't access a shared object file to preload it, so it just ignores the variable. Checked that the path was correct, checked permissions, it all checks out, so not sure what's going on here. Starting to look like it's more a "me" issue than an AM2R issue though.

`ERROR: ld.so: object '/usr/lib/i386-linux-gnu/libGL.so.1.' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
ERROR: ld.so: object '/usr/lib/i386-linux-gnu/libGL.so.1.' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
ERROR: ld.so: object '/usr/lib/i386-linux-gnu/libGL.so.1.' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
`

**Comment by Lojemiru** [comment]
You should check to see if you have 32-bit support enabled using `dpkg --print-foreign-architectures`.

**Comment by LordSicarious** [comment]
Yeah, i386 is enabled. Genuinely don't know what's going on with this ATM, it's weird as hell.

**Comment by Lojemiru** [comment]
Hm. Out of my depth, we'll see what @Miepee has to say.

**Comment by Miepee** [comment]
> `ERROR: ld.so: object '/usr/lib/i386-linux-gnu/libGL.so.1.' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored. ERROR: ld.so: object '/usr/lib/i386-linux-gnu/libGL.so.1.' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored. ERROR: ld.so: object '/usr/lib/i386-linux-gnu/libGL.so.1.' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored. `
That error means, you don''t have the 32-Bit versions of your driver installed.
https://www.nvidia.com/Download/driverResults.aspx/156162/en-us

**Comment by TauAkiou** [comment]
For NVIDIA drivers, try using  ```LD_PRELOAD=/usr/lib/libGLX_nvidia.so.0```.

**Comment by ChristianSilvermoon** [comment]
Can confirm that the issue occurs for me as well, on Linux Mint 20.2

**Comment by Miepee** [comment]
Did you try the suggestion(s) from above?

**Comment by ChristianSilvermoon** [comment]
> Did you try the suggestion(s) from above?

Been working on it.
Using `LD_PRELOAD=/usr/lib/i386-linux-gnu/libGLX_nvidia.so.0`

does get the game to launch :)

I'll try adding that to the launcher environment variables

**Comment by Duder963** [comment]
I can report that on Arch Linux, with an NVIDIA GTX 1080 Ti, using LD_PRELOAD=/usr/lib32/libGLX_nvidia.so allows the game to run. However, audio doesn't seem to work. Executing AM2R.AppImage from the terminal shows a ton of OpenAL errors. I am using pipewire, so that could be the issue.

**Comment by Miepee** [comment]
@Duder963 late response, but double check that you have `lib32-libpulse` and `lib32-openal` installed.

**Comment by AverageGabe** [comment]
Same problem as the original post, I'm on Arch Linux with an Nvidia 3070. I've tried all of the solutions above but none of them have worked

**Comment by Miepee** [comment]
@AverageGabe can you try the gpu-fix branch?

**Comment by AverageGabe** [comment]
I ended up fixing it right after my previous post by using `LD_PRELOAD=/usr/lib32/libGL.so`

**Comment by Miepee** [comment]
Closed by new 2.2.0 AM2RLauncher release / merging of gpu-fix on Autopatcher-Linux.
Delete and reinstall the profile (go to the mod settings tab, open the `open profile content folder` button, delete the folder it opens`).
Should you still have issues, feel free to reopen it.
